apps/gitness/src/pages-v2/create-project.tsx

```diff
@@ -0,0 +1,45 @@
+import { useNavigate } from 'react-router-dom'
+
+import { useCreateSpaceMutation } from '@harnessio/code-service-client'
+import { CreateProjectFields, CreateProjectPage } from '@harnessio/ui/views'
+
+import { useAppContext } from '../framework/context/AppContext'
+import { useRoutes } from '../framework/context/NavigationContext'
+import { useTranslationStore } from '../i18n/stores/i18n-store'
+
+export const CreateProject = () => {
+  const routes = useRoutes()
+  const navigate = useNavigate()
+  const { addSpaces, spaces } = useAppContext()
+
+  const isAdditional = spaces.length >= 1
+
+  const {
+    mutate: createSpace,
+    isLoading,
+    error
+  } = useCreateSpaceMutation(
+    {},
+    {
+      onSuccess: response => {
+        const { body: project } = response
+        addSpaces([project])
+        navigate(routes.toRepositories({ spaceId: project?.identifier }))
+      }
+    }
+  )
+
+  const handleFormSubmit = (formData: CreateProjectFields) => {
+    createSpace({ body: { identifier: formData.name, description: formData.description } })
+  }
+
+  const commonProps = { isLoading, error: error?.message, onFormSubmit: handleFormSubmit, useTranslationStore }
+
+  return (
+    <>
+      {isAdditional && <CreateProjectPage {...commonProps} isAdditional backLinkProps={{ to: routes.toHome() }} />}
+
+      {!isAdditional && <CreateProjectPage {...commonProps} logoutLinkProps={{ to: routes.toLogout() }} />}
+    </>
+  )
```

Comment:
Same here, maybe a bit easier to do the conditional outside of JSX.
```suggestion
  if (isAdditional) {
    return <CreateProjectPage {...commonProps} isAdditional backLinkProps={{ to: routes.toHome() }} />
  }

  return <CreateProjectPage {...commonProps} logoutLinkProps={{ to: routes.toLogout() }} />
```

Or even a ternary (though I prefer the obviousness of the `if`)
```suggestion
  return isAdditional
    ? <CreateProjectPage {...commonProps} isAdditional backLinkProps={{ to: routes.toHome() }} />
    : <CreateProjectPage {...commonProps} logoutLinkProps={{ to: routes.toLogout() }} />
```